modules/stochastic_tools/include/state_sim/common/StateSimIdAndDesc.h

```diff
@@ -0,0 +1,30 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMIDANDDESC_H
+#define STATESIMIDANDDESC_H
+
+#include <string>
+
+/**
+ * Simple class to store the path of state movement and the reason for that movement.
+ */
+class StateSimIdAndDesc
+{
+public:
+  /**
+   * This is the main construter for the user.
+   * @param in_id - item id
+   * @param in_desc - description.
+   */
+  StateSimIdAndDesc(int in_id, std::string in_desc) : id(in_id), desc(in_desc) {};
```

Comment:
Our coding conventions state that each member is initialized on it's own line. Also, all member variables should have a leading underscore. We do make exceptions for structures though which is basically what this is. I'd prefer to see this as a struct if you want to avoid using underscores on the members.

Also, you have an extra trailing semicolon.